### PR TITLE
Add start and stop to AWS instances

### PR DESCRIPTION
**Pull request by RazTamir**

### CODE REVIEW

File: utility/aws.py [hunk 1]
```diff
@@ -238,3 +242,35 @@ def detach_and_delete_volume(self, volume, timeout=120):
             "Delete response for volume: %s is: %s", ec2_volume.volume_id,
             delete_response
         )
+
+    def stop_node(self, instance_id):
```
Comment (petr-balogh) [inline utility/aws.py]: rename to stop_instance or stop_ec2_instance please

File: utility/aws.py [hunk 2]
```diff
@@ -238,3 +242,35 @@ def detach_and_delete_volume(self, volume, timeout=120):
...
+            instance_id (str): ID of the instance to stop
+
+        Returns:
+            bool: True in case operation succeeded, False otherwise
+        """
+        res = self.ec2_client.stop_instances(
+            InstanceIds=[instance_id], Force=True
+        )
+        state = res.get('StoppingInstances')[0].get('CurrentState').get('Code')
+        return state == STOPPING
```
Comment (petr-balogh) [inline utility/aws.py]: No need to wait till instance is really stopped?

File: utility/aws.py [hunk 3]
```diff
@@ -238,3 +242,35 @@ def detach_and_delete_volume(self, volume, timeout=120):
...
+
+    def start_node(self, instance_id):
```
Comment (petr-balogh) [inline utility/aws.py]: start_instance or start_ec2_instance

File: utility/aws.py [hunk 4]
```diff
@@ -238,3 +242,35 @@ def detach_and_delete_volume(self, volume, timeout=120):
...
+            instance_id (str): ID of the instance to start
+
+        Returns:
+            bool: True in case operation succeeded, False otherwise
+        """
+        res = self.ec2_client.stop_instances(
+            InstanceIds=[instance_id], Force=True
+        )
+        state = res.get('StartingInstances')[0].get('CurrentState').get('Code')
+        return state == PENDING
```
Comment (petr-balogh) [inline utility/aws.py]: No need for wait till instance is started?

File: utility/aws.py (hunk 2 above)
Comment (RazTamir) [reply utility/aws.py]: done

File: utility/aws.py (hunk 1 above)
Comment (RazTamir) [reply utility/aws.py]: done

File: utility/aws.py (hunk 3 above)
Comment (RazTamir) [reply utility/aws.py]: done

File: utility/aws.py (hunk 4 above)
Comment (RazTamir) [reply utility/aws.py]: done

File: utility/aws.py [hunk 5]
```diff
@@ -104,6 +114,20 @@ def get_instances_by_name_pattern(self, pattern):
...
+
+        Args:
+            instance_id (str): ID of the instance
+
+        Returns:
+            str: The instance status
+        """
+        return self.ec2_client.describe_instances(
+            InstanceIds=[instance_id],
+        )['Reservations']['Instances'][0]['State']['Code']
```
Comment (ebenahar) [inline utility/aws.py]: used .get()

File: utility/aws.py [hunk 6]
```diff
@@ -238,3 +262,53 @@ def detach_and_delete_volume(self, volume, timeout=120):
...
+
+        Args:
+            instance_id (str): ID of the instance to start
+            wait (bool): True in case wait for status is needed,
+                False otherwise
+
+        Returns:
+            bool: True in case operation succeeded, False otherwise
+        """
+        res = self.ec2_client.stop_instances(
```
Comment (ebenahar) [inline utility/aws.py]: change to start_instances()
Is there a need for 'Force' arg in start?

File: utility/aws.py (hunk 5 above)
Comment (RazTamir) [reply utility/aws.py]: done

File: utility/aws.py (hunk 6 above)
Comment (RazTamir) [reply utility/aws.py]: done

**Comment by RazTamir** [comment]
No way to prove it was verified but it is

**Pull request merged**